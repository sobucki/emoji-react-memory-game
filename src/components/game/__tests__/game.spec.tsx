import { fireEvent, render, screen, within } from "@testing-library/react";
import Game from "..";

const renderGame = () => {
  return render(<Game />);
};

describe("game", () => {
  describe("rendering", () => {
    it("should match snapshot", () => {
      const { asFragment } = renderGame();
      expect(asFragment()).toMatchSnapshot();
    });

    it("should start with menu opened", () => {
      renderGame();
      const modalTitle = screen.getByText("Start a new game");
      expect(modalTitle).toBeInTheDocument();
    });
  });

  describe("behavior", () => {
    it("should close menu after input and press start", () => {
      renderGame();

      const categorySelect = screen.getByRole("combobox", {
        name: /category:/i,
      });
      fireEvent.change(categorySelect, { target: { value: "animals_nature" } });

      const levelSelect = screen.getByRole("combobox", { name: /level:/i });
      fireEvent.change(levelSelect, { target: { value: "normal" } });

      fireEvent.click(screen.getByRole("button", { name: /start/i }));

      expect(screen.queryByText("Start a new game")).not.toBeInTheDocument();
    });

    it.only("should start a game with VERY EASY level", () => {
      renderGame();

      const categorySelect = screen.getByRole("combobox", {
        name: /category:/i,
      });
      const categoryOptions = within(categorySelect).queryAllByRole("option");

      const firstOption = categoryOptions.find(
        (option) => (option as HTMLOptionElement).value
      ).?;
      console.log(firstOption.val);

      // expect(categoryOptions).toEqual([]);
    });
  });
});
