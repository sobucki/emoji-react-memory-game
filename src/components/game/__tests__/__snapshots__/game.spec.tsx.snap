// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > rendering > should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-dmyCSP fbZkES"
  >
    <div>
      time: 00:00
    </div>
    <div>
      moves: 0
    </div>
    <div>
      time
    </div>
  </section>
  <div
    class="sc-guDLey iHIfwE"
  />
  <div
    class="sc-hLQSwg bKgnDw"
  >
    <div
      class="sc-eDLKkx kcLWBJ"
    >
      <h2>
        Start a new game
      </h2>
      <fieldset>
        <legend>
          Configure how you want to play
        </legend>
        <form>
          <label
            for="category-select"
          >
            Category:
          </label>
          <select
            id="category-select"
            name="categories"
            required=""
          >
            <option
              value=""
            >
              --Please choose an category--
            </option>
            <option
              value="activity_sports"
            >
              Activity and Sports ⚽️🤸‍♂️
            </option>
            <option
              value="animals_nature"
            >
              Animals & Nature 🐶 🐱
            </option>
            <option
              value="clothing_accessories"
            >
              Clothing and Accessories 🧦🕶
            </option>
            <option
              value="food_drink"
            >
              Food & Drink 🍕🍺
            </option>
            <option
              value="gestures_body"
            >
              Gestures and Body Parts 👋🦶
            </option>
            <option
              value="objects"
            >
              Objects 🔧💡
            </option>
            <option
              value="people_fantasy"
            >
              People and Fantasy 👨🧛‍♂️
            </option>
            <option
              value="smileys"
            >
              Smileys 😃😝
            </option>
            <option
              value="flags"
            >
              Flags 🇧🇷🇺🇸
            </option>
            <option
              value="symbols"
            >
              Symbols 🆘🛂
            </option>
            <option
              value="travel_places"
            >
              Travel & Places 🚙 🌆
            </option>
          </select>
          <label
            for="level-select"
          >
            Level:
          </label>
          <select
            id="level-select"
            name="levels"
            required=""
          >
            <option
              value=""
            >
              --Please choose an level--
            </option>
            <option
              value="very_easy"
            >
              🤩 Very easy (5 emojis)
            </option>
            <option
              value="easy"
            >
              🤗 Easy (8 emojis)
            </option>
            <option
              value="normal"
            >
              🙂 Normal (10 emojis)
            </option>
            <option
              value="hard"
            >
              😣 Hard (15 emojis)
            </option>
            <option
              value="very_hard"
            >
              😨 Very hard (20 emojis)
            </option>
            <option
              value="insane"
            >
              😱 Insane (50 emojis)
            </option>
          </select>
          <button
            type="submit"
          >
            Start
          </button>
        </form>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
